nzedb/processing/tv/TMDB.php

```diff
@@ -0,0 +1,347 @@
+		return $return;
+	}
+
+	/**
+	 * @param $showArr
+	 * @param $cleanName
+	 *
+	 * @return array|bool
+	 */
+	private function matchShowInfo($showArr, $cleanName)
+	{
+		$return = false;
+		$highestMatch = 0;
+
+		foreach ($showArr AS $show) {
+			if ($this->checkRequired($show->_data, 'tmdbS')) {
+				// Check for exact title match first and then terminate if found
+				if (strtolower($show->_data['name']) === strtolower($cleanName)) {
+					$highest = $show;
+					break;
+				} else {
+					// Check each show title for similarity and then find the highest similar value
+					$matchPercent = $this->checkMatch(strtolower($show->_data['name']), strtolower($cleanName), self::MATCH_PROBABILITY);
+
+					// If new match has a higher percentage, set as new matched title
+					if ($matchPercent > $highestMatch) {
+						$highestMatch = $matchPercent;
+						$highest = $show;
+					}
+				}
+			}
+		}
+		if (isset($highest)) {
+			$showAppends = $this->client->getTVShow($highest->_data['id'], 'append_to_response=alternative_titles,external_ids');
+			if ($showAppends) {
+				foreach ($showAppends->_data['alternative_titles']['results'] AS $aka) {
+					$highest->_data['alternative_titles'][] = $aka['title'];
+				}
+				$highest->_data['network'] = (isset($showAppends->_data['networks'][0]['name']) ? $showAppends->_data['networks'][0]['name'] : '');
+				$highest->_data['external_ids'] = $showAppends->_data['external_ids'];
+			}
+			$return = $this->formatShowArr($highest);
+		}
+		return $return;
+	}
+
+	/**
+	 * Retrieves the poster art for the processed show
+	 *
+	 * @param int $videoId -- the local Video ID
+	 * @param int $showId  -- the TMDB ID
+	 *
+	 * @return null
+	 */
+	protected function getPoster($videoId, $showId = 0)
+	{
+		$ri = new ReleaseImage($this->pdo);
+
+		// Try to get the Poster
+		$hascover = $ri->saveImage($videoId, sprintf($this->posterUrl), $this->imgSavePath, '', '');
+
+		// Mark it retrieved if we saved an image
+		if ($hascover == 1) {
+			$this->setCoverFound($videoId);
+		}
+	}
+
+	/**
+	 * Gets the specific episode info for the parsed release after match
+	 * Returns a formatted array of episode data or false if no match
+	 *
+	 * @param integer $tmdbid
+	 * @param integer $season
+	 * @param integer $episode
+	 * @param string  $airdate
+	 * @param integer $videoId
+	 *
+	 * @return array|bool
+	 */
+	protected function getEpisodeInfo($tmdbid, $season, $episode, $airdate = '', $videoId = 0)
+	{
+		$return = false;
+
+		$response = $this->client->getEpisode($tmdbid, $season, $episode);
+
+		sleep(1);
+
+		//Handle Single Episode Lookups
+		if (is_object($response)) {
+			if ($this->checkRequired($response->_data, 'tmdbE')) {
+				$return = $this->formatEpisodeArr($response);
+			}
+		}
+		return $return;
+	}
+
+	/**
+	 * Assigns API show response values to a formatted array for insertion
+	 * Returns the formatted array
+	 *
+	 * @param $show
+	 *
+	 * @return array
+	 */
+	private function formatShowArr($show)
```

Comment:
Okay, I will try to do that and not fuck it up.  It's TV specific, so I will put it there.  If I create something abstract in TV, does it need to exist in Videos as well as the site scraper classes?